src/block.c

```diff
@@ -258,4 +259,1694 @@ coap_add_data_blocked_response(coap_resource_t *resource,
                 (const unsigned char *)coap_response_phrase(response->code));
 }
 
-#endif /* WITHOUT_BLOCK  */
+void coap_context_set_block_mode(coap_context_t *context,
+                                  uint8_t block_mode) {
+  context->block_mode = block_mode &= (COAP_BLOCK_USE_LIBCOAP |
+                                       COAP_BLOCK_SINGLE_BODY);
+  if (!(block_mode & COAP_BLOCK_USE_LIBCOAP))
+    context->block_mode = 0;
+}
+
+/*
+ * The block token match only matches on the bottom 32 bits
+ * [The upper 32 bits are incremented as different payloads are sent]
+ *
+ */
+COAP_STATIC_INLINE int
+block_token_match(const uint8_t *a, size_t alen,
+  const uint8_t *b, size_t blen) {
+  if (blen < 4)
+    return alen == blen && memcmp(a, b, blen) == 0;
+  uint32_t bias = blen - 4;
```

Comment:
For more portability, the definition of `bias` should come before the first instruction.

Reply:
Moved `bias`.

Reply:
Ah, this should result at least in a warning. The initialization should come after the length check:

```c
uint32_t bias;
if (blen < 4) ....
bias = blen -4 ;
...
```

Reply:
Code updated